src/Nox.Solution/Validation/Domain/OwnedEntityValidator.cs

```diff
@@ -0,0 +1,51 @@
+﻿
+using FluentValidation;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nox.Solution.Validation;
+
+internal class OwnedEntityValidator : AbstractValidator<Entity>
+{
+    public OwnedEntityValidator(IEnumerable<Entity> entities)
+    {
+        /*
+            1. Owned entity with ZeroOrOne or ExactlyOne relationship keys must be null. 
+            2. Keys are mandatory for all other entity usages
+            3. Keys must be single for Entity used a Foreign Key 
+        */
+        When(e => e.Relationships.Any(), () =>
+        {
+            When(e => e.Relationships.Any(x => x.Relationship == EntityRelationshipType.ZeroOrOne || x.Relationship == EntityRelationshipType.ExactlyOne), () =>
+            {
+                RuleFor(e => e.Keys)
+                    .Empty()
+                    .WithMessage(e => string.Format(ValidationResources.OwnedEntityKeysMustBeNull, e.Name));
+            })
+            .Otherwise(() =>
+            {
+                RuleFor(e => e.Keys)
+                   .NotEmpty()
+                   .WithMessage(e => string.Format(ValidationResources.EntityKeysRequired, e.Name));
+            });
+        });
+
+        // Owned entity cannot be auditable
+        RuleFor(e => e.Persistence!.IsAudited)
+            .NotEqual(true)
+            .WithMessage(e => string.Format(ValidationResources.EntityOwnedCannotBeAuditable, e.Name));
+
+        // Owned entity cannot have relationships to other entities
+        RuleFor(e => e.Relationships.Any())
```

Comment:
FluentValidation evaluates the rules in the order they are defined and stops as soon as it encounters a validation failure, so please move this validation rule to the first one in the OwnedEntityValidator.

Reply:
The default behavior of the validator is to run all of the rules. If we want the validator to stop on first failure, then we would need to specify the validator-level casccade mode. https://docs.fluentvalidation.net/en/latest/cascade.html